include/oneapi/dpl/internal/async_impl/async_impl_hetero.h

```diff
@@ -170,15 +165,14 @@ __pattern_transform_reduce_async(_ExecutionPolicy&& __exec, _ForwardIterator __f
     auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _ForwardIterator>();
     auto __buf = __keep(__first, __last);
 
-    auto res = oneapi::dpl::__par_backend_hetero::__parallel_transform_reduce_async<_RepackedTp>(
+    auto __res = oneapi::dpl::__par_backend_hetero::__parallel_transform_reduce<_RepackedTp>(
         ::std::forward<_ExecutionPolicy>(__exec),
         unseq_backend::transform_init<_Policy, _BinaryOperation, _Functor>{__binary_op,
                                                                            _Functor{__unary_op}}, // transform
         __binary_op,                                                                              // combine
         unseq_backend::reduce<_Policy, _BinaryOperation, _RepackedTp>{__binary_op},               // reduce
         __buf.all_view());
-    res.set(__init);
-    return res;
+    return oneapi::dpl::__internal::__future<_Tp>(::std::move(__res), __init, __binary_op);
```

Comment:
Could you please clarify why `std::move` is uesd here, not just copy?

Reply:
Move constructor is used to enable stealing resources from internal future (e.g. unique pointer to keep alive objects)

Reply:
ok. So, we don't need to access `__res` that's why we can steal resources.